lib/logstash/util/accessors.rb

```diff
@@ -48,44 +47,37 @@ def strict_set(accessor, value)
 
     def del(accessor)
       target, key = lookup(accessor)
-      unless target.nil?
-        target.is_a?(Array) ? target.delete_at(key.to_i) : target.delete(key)
-      end
+      return nil unless target
+      target.is_a?(Array) ? target.delete_at(key.to_i) : target.delete(key)
     end
 
     def include?(accessor)
-      target, key = lookup_path(accessor)
+      target, key = lookup(accessor)
```

Comment:
this eagerness to be `nil` is still something I find difficult to swallow in Ruby, but I get it.

```
x = nil
y, z = nil
y.nil? && z.nil? #true
```